src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php

```diff
@@ -12,46 +12,59 @@
 namespace Sylius\Bundle\TaxonomyBundle\Doctrine\ORM;
 
 use Sylius\Bundle\TranslationBundle\Doctrine\ORM\TranslatableResourceRepository;
-use Sylius\Component\Taxonomy\Model\TaxonomyInterface;
+use Sylius\Component\Taxonomy\Model\TaxonInterface;
 use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface;
 
 /**
- * @author Saša Stamenković <umpirsky@gmail.com>
- * @author Gonzalo Vilaseca <gvilaseca@reiss.co.uk>
- * @author Anna Walasek <anna.walasek@lakion.com>
+ * @author Aram Alipoor <aram.alipoor@gmail.com>
  */
 class TaxonRepository extends TranslatableResourceRepository implements TaxonRepositoryInterface
 {
-    public function getTaxonsAsList(TaxonomyInterface $taxonomy)
+    /**
+     * @param TaxonInterface $taxon
+     *
+     * @return array
+     */
+    public function findChildren(TaxonInterface $taxon)
     {
-        return $this->getQueryBuilder()
-            ->where('o.taxonomy = :taxonomy')
-            ->andWhere('o.parent IS NOT NULL')
-            ->setParameter('taxonomy', $taxonomy)
-            ->orderBy('o.left')
-            ->getQuery()
-            ->getResult();
+        $queryBuilder = $this->getQueryBuilder();
+        $queryBuilder
+            ->andWhere($queryBuilder->expr()->gt('o.left', ':left'))
+            ->andWhere($queryBuilder->expr()->lt('o.right', ':right'))
+            ->setParameter('left', $taxon->getLeft())
+            ->setParameter('right', $taxon->getRight())
+        ;
+
+        return $queryBuilder->getQuery()->getResult();
     }
 
+    /**
+     * @param string $permalink
+     *
+     * @return TaxonInterface
+     *
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     */
     public function findOneByPermalink($permalink)
```

Comment:
Since you implement TaxonRepositoryInterface this function should be defined there. You should move docblocks also.